gitpod-io/gitpod
# Integrate reading from log files into Supervisor's `terminal.Listen`

## Description

Make it possible to access logs of complete prebuild tasks even before they are backed up and served from s3 / GCP storage.

## Related Issue(s)

Fixes ENT-224

## How to test

1. [Join my org](https://ft-supervi52cbd2f4e2.preview.gitpod-dev.com/orgs/join?inviteId=5fb6722d-5b33-4cce-a817-8aa839691c77)
2. Run a prebuild on a repo with multiple prebuild tasks, such as the already imported https://github.com/filiptronicek/test-gitpod-tasks-with-variable-lengths or https://github.com/filiptronicek/gitpod (I forked Gitpod, because I had to have rights on the repo to enable prebuilds)
3. Observe all tasks' logs are accessible in all states

## Documentation


#### Preview status

<p>Gitpod was successfully deployed to your preview environment.</p>
<ul>
	<li><b>🏷️ Name</b> - ft-supervi52cbd2f4e2</li>
	<li><b>🔗 URL</b> - <a href="https://ft-supervi52cbd2f4e2.preview.gitpod-dev.com/workspaces" target="_blank">ft-supervi52cbd2f4e2.preview.gitpod-dev.com/workspaces</a>.</li>
	<li><b>📚 Documentation</b> - See our <a href="https://www.notion.so/gitpod/6debd359591b43688b52f76329d04010#7c1ce80ab31a41e29eff2735e38eec39" target="_blank">internal documentation</a> for information on how to interact with your preview environment.</li>
	<li><b>📦 Version</b> - ft-supervisor-listen-files-gha.26079</li>
	<li><b>🗒️ Logs</b> - <a href="https://console.cloud.google.com/logs/query;query=jsonPayload.kubernetes.host%3D%22preview-ft-supervi52cbd2f4e2%22%0A%0A--%20Filter%20on%20service:%0A--%20jsonPayload.serviceContext.service%3D%22ws-manager-mk2%22%0A;duration=P1D?project=gitpod-dev-preview" target="_blank">GCP Logs Explorer</a></li>
</ul>

## Build Options

<details>
<summary>Build</summary>

      Run the build with werft instead of GHA
      Run Leeway with `--dont-test`
</details>

<details>
<summary>Publish</summary>

</details>

<details>
<summary>Installer</summary>

  Add desired feature flags to the end of the line above, space separated
</details>

<details>
<summary>Preview Environment / Integration Tests</summary>

      If enabled this will build `install/preview`
      If enabled this will create the environment on GCE infra
      Saves cost. Untick this only if you're really sure you need a non-preemtible machine.
      Valid options are `all`, `workspace`, `webapp`, `ide`, `jetbrains`, `vscode`, `ssh`. If enabled, `with-preview` and `with-large-vm` will be enabled.
</details>

/hold
